lang/cs/Org.Apache.REEF.Client/YARN/RESTClient/IHttpClient.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.YARN.RestClient
+{
+    /// <summary>
+    /// Wrapper interface on <see cref="System.Net.Http.HttpClient"/> for ease of unit-testing
+    /// </summary>
+    [DefaultImplementation(typeof(HttpClient))]
+    internal interface IHttpClient
```

Comment:
Would we have multiple implementations of this interface? Or are we just using this for unit tests? If the latter, I'm not sure if we should actually be writing tests for `System.Net.Http.HttpClient`.